src/index.mjs

```diff
@@ -0,0 +1,220 @@
+// eslint-disable-next-line import/no-unresolved
+import tseslint from 'typescript-eslint';
+import globals from 'globals';
+import unicorn from 'eslint-plugin-unicorn';
+import sonarjs from 'eslint-plugin-sonarjs';
+import prettier from 'eslint-plugin-prettier/recommended';
+import jest from 'eslint-plugin-jest';
+import importPlugin from 'eslint-plugin-import';
+import js from '@eslint/js';
+
+export default [
+  js.configs.recommended,
+  ...tseslint.configs.recommended,
+  sonarjs.configs.recommended,
+  {
+    ignores: ['dist', 'coverage/*'],
+  },
+  {
+    languageOptions: {
+      globals: {
+        ...globals.node,
+        ...globals.es2020,
+        ...globals.builtin,
+      },
+      parserOptions: {
+        ecmaVersion: 2020,
+        sourceType: 'module',
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
+    plugins: {
+      import: importPlugin,
+      jest,
+      unicorn,
+    },
+    rules: {
+      'sonarjs/max-switch-cases': ['error', 15],
+      'sonarjs/no-empty-function': 'off',
+      'sonarjs/no-unused-expressions': 'off',
+      'sonarjs/todo-tag': 'off',
+      'sonarjs/fixme-tag': 'off',
+      'sonarjs/jsx-no-constructed-context-values': 'warn',
+      'sonarjs/no-array-index-key': 'warn',
+      'sonarjs/unused-import': 'off',
+      'no-console': 'error',
+      'import/order': [
+        'error',
+        {
+          'newlines-between': 'always-and-inside-groups',
+          groups: [
+            ['builtin', 'external'],
+            'internal',
+            ['parent', 'sibling'],
+            ['index', 'object'],
+          ],
+          alphabetize: {
+            order: 'desc',
+            caseInsensitive: true,
+          },
+        },
+      ],
+      'import/no-unresolved': 'error',
+      'sort-imports': [
+        'error',
+        {
+          ignoreCase: true,
+          ignoreDeclarationSort: true,
+          ignoreMemberSort: false,
+          memberSyntaxSortOrder: ['none', 'single', 'all', 'multiple'],
+          allowSeparatedGroups: true,
+        },
+      ],
+      '@typescript-eslint/no-unused-vars': 'off',
+      '@typescript-eslint/no-require-imports': 'off',
+      'no-unused-vars': [
+        'error',
+        {
+          varsIgnorePattern: '^_',
+          argsIgnorePattern: '^_',
+          ignoreRestSiblings: true,
+        },
+      ],
+      'no-shadow': 'error',
```

Comment:
Wasn't this duplicated by `typescript-eslint` or `sonarjs`?

Reply:
for ts files `no-shadow` is turned off and `typescript/no-shadow` enabled (line 165)